ENT-1987 Take 2

muhammad-ammar opened the pull request:
**Description:** This updates the `Enterprise Customer Details` link below `Enterprise Customer Catalog` field on Enterprise Coupon Detail/Edit screen. 

**JIRA:** [ENT-1987](https://openedx.atlassian.net/browse/ENT-1987)

**Testing:** PR changes are deployed to sandbox.

**Admin Credentials:**
username: edx 
password: edx

1. Login to https://ecommerce-mammar.sandbox.edx.org/login
2. Coupon Edit Screen: [Enterprise Catalog Details link](https://ecommerce-mammar.sandbox.edx.org/enterprise/coupons/7/edit/) should work
3. Coupon Detail Screen: [Enterprise Catalog Details link](https://ecommerce-mammar.sandbox.edx.org/enterprise/coupons/7/) should work

brittneyexline commented on ecommerce/extensions/api/v2/views/enterprise.py:
```diff
@@ -84,6 +89,17 @@ def get(self, request):
 
+    def retrieve(self, request, **kwargs):
+        endpoint_request_url = urlparse(request.build_absolute_uri())._replace(query=None).geturl()
+        catalog = get_enterprise_catalog(
```
you should catch potential exceptions here and return a formatted error message so that there is some insight into any failing requests

muhammad-ammar commented:
@brittneyexline The 500 is coming because implicit access is not working on sandbox because oauth authentication is not working. Only the explicit access is currently working on my sandbox and linking the ecommerce_worker user with multiple enterprises will not work because we are explicitly checking the multiple enterprises [here](https://github.com/edx/edx-enterprise/blob/master/enterprise/models.py#L1805-L1828)

I will add the exception handling and see why empty results are returned?

brittneyexline commented:
@muhammad-ammar - the original reason we needed to change our approach on this is because this case needs to work for implicit access specifically; this page is used by the ECS team who are internal edx admins. this page would never be exposed to a non-edx employee on our instance. if i manage to make progress on the business sandbox today, then we can test there first. otherwise, if we're having trouble testing, we can release to stage and test there.

muhammad-ammar commented:
@brittneyexline Authentication is fixed on sandbox and all the catalog related issues are fixed now. Please have a look again.

brittneyexline commented on ecommerce/extensions/api/v2/views/enterprise.py:
```diff
@@ -84,6 +91,29 @@ def get(self, request):
+                exc
+            )
+            return Response(
+                {'error': 'Unable to retrieve enterprise catalog'},
```
i think we can surface the exception message here as well

muhammad-ammar commented:
@brittneyexline Addressed your feedback. Any idea how to render indented catalog response when  Enterprise Catalog Details link is opened in the new tab? We already have `JSONRenderer` as our default renderer in ecommerce but json is not indented. I have also tried the `indent` query param but no luck.

brittneyexline commented:
@muhammad-ammar - not sure if there is much you can do from the code side, i think rendering json is the right thing to do. i have a chrome extension that formats json so that would be my recommendation to those utilizing this link.

muhammad-ammar commented:
@brittneyexline BTW firefox implicitly beautify the json response without any extension.
